docs/guides/using-tls-mutual-authentication.md

```diff
@@ -0,0 +1,103 @@
+
+mTLS adds an additional flow to the previous TLS server verification steps.  The client will still verifiy the server's identity, but now the server will in turn also verify the identity of the client.  By verifying the clients, the server owner is able to restrict access only to verified clients strengthening security.
+![](img/mtls-diag.png)
+
+This document will provide a quick start guide for how to enable mTLS on your ngrok hosted endpoints.  For a deeper understanding for how mTLS is implemented within ngrok, reference the [Mutual TLS module page.](https://ngrok.com/docs/cloud-edge/modules/mutual-tls/)
+
+
+## **Prerequisites**
+
+A certificate authority (CA) is required for mTLS.  The CA is responsible for issuing and digitally signing certificates (client certificates).  The CA will also be used to verify the authenticity of the certificates.  
+
+1. CA certificate to be used by ngrok to verify the clients
+2. Client certificates signed by the CA used to access the endpoints
+
+Most organizations will have their own certificate mangement infrastructure, examples are provided below for creating the CA and client certificates.  These examples of certificate creation are optional, and provided for reference/demo purposes.
+
+### Example - Create the CA
+
+Generate the CA private key, providing a strong passphrase is desirable<br></br>
+```bash
+openssl genrsa -aes256 -out ca.key 4096
+```
+
+
+Now we'll generate the CA certificate.  Feel free to leave attribute fields blank by entering a period `.`.  It's recommended to specify the CommonName this will be used to identify the certificate authority.<br></br>
+```bash
+openssl req -new -x509 -sha256 -days 365 -key ca.key -out ca.crt
+```
+
+### Example - Create a Client Certificate
+
+Generate the client private key<br></br>
+```bash
+openssl genrsa -out ngrokclient.key 2048
+```
+
+Create a signing request for the client private key<br></br>
+```bash
+openssl req -new -key ngrokclient.key -out ngrokclient.csr
+```
+
+Use the CA certificate to sign the request and generate the client certificate<br></br>
+```bash
+openssl \
+   x509 -req -days 365 -sha256 -in  ngrokclient.csr \
+   -CA ca.crt -CAkey ca.key -set_serial 0x"$(openssl rand -hex 16)" -out ngrokclient.crt
+```
+
+
+## **Enable mTLS**
+With the prerequisites being met, let's see how simple it is to enable mTLS with ngrok.  Examples for using either the Agent CLI or ngrok Cloud Edges are provided below.
+
+### Agent CLI
+Specify the CA certificate when starting the tunnel.<br></br>
```

Comment:
this seems a little short. maybe add a paragraph below as to what this command is doing? for example, 'this command uploads the CA to the ngrok saas platform and then uses it to verify requests?'